src/ReinforcementLearningCore/src/core/hooks.jl

```diff
@@ -286,26 +286,26 @@ function Base.push!(hook::TimePerStep, ::PostActStage, agent, env)
 end
 
 """
-    DoEveryNStep(f; n=1, t=0)
+    DoEveryNSteps(f; n=1, t=0)
 
 Execute `f(t, agent, env)` every `n` step.
 `t` is a counter of steps.
 """
-mutable struct DoEveryNStep{F,T} <: AbstractHook where {F,T<:Integer}
+mutable struct DoEveryNSteps{F,T} <: AbstractHook where {F,T<:Integer}
     f::F
     n::T
     t::T
 
-    function DoEveryNStep(f; n=1, t=0)
+    function DoEveryNSteps(f; n=1, t=0)
         new{typeof(f),Int64}(f, n, t)
     end
 
-    function DoEveryNStep{T}(f; n=1, t=0) where {T<:Integer}
+    function DoEveryNSteps{T}(f; n=1, t=0) where {T<:Integer}
```

Comment:
Couldn't we have 
```
mutable struct DoEveryNSteps{F} <: AbstractHook where {F}
    f::F
    n::Int
    t::Int
end
```
instead of supporting multiple Int precisions ? This would make the last constructor unnecessary. I feel like this is extra code to compile that noone will ever need and which may lead to sneaky bugs.